### PR TITLE
sparse: Make sure ro.build.user and ro.build.host is always the same if replacements are specified.

### PR DESCRIPTION
**Pull request by krnlyng**
[sparse] Make sure ro.build.user and ro.build.host is always the same. JB#54990

### CODE REVIEW

File: helpers/copy_tree.sh [hunk 1]
```diff
@@ -46,6 +62,9 @@ sed -i "/$token/,\$d" "$modify_spec"
 echo "$token" >> "$modify_spec"
 echo "%post" >> "$modify_spec"
 
+BUILD_HOST_REPLACEMENT=$(grep "%define build_host_replacement" "$modify_spec" | cut -d ' ' -f3)
+BUILD_USER_REPLACEMENT=$(grep "%define build_user_replacement" "$modify_spec" | cut -d ' ' -f3)
+
```
Comment (sledges) [inline helpers/copy_tree.sh:67]: Let's make [this](https://github.com/mer-hybris/droid-system-device/pull/22/files#diff-491d059e29b8e3d61f1fd32e89ae18b2e3a499f1a7f5fa6e34506d73e8f510e1R6) fool-proof:)
```suggestion

if [ -n "$BUILD_HOST_REPLACEMENT" ] && [[ ${trees[0]} != *system ]]; then
    echo Make sure to pass .../system argument first to this script, this is needed for
    echo removing the ro.build.host and ro.build.user properly from .../vendor.
    exit    
fi

```

File: helpers/copy_tree.sh [hunk 2]
```diff
@@ -27,6 +29,20 @@ get_closest() {
     fi
 }
 
+BUILD_HOST=
+BUILD_USER=
+fixup_buildprop() {
+    if [ -f "$1" ]; then
+        # pray that these are unique enough.
+        if [ ! -z "$BUILD_HOST" ] && [ ! -z "$BUILD_HOST_REPLACEMENT" ]; then
```
Comment (sledges) [inline helpers/copy_tree.sh:37]: shellcheck prefers this to be `-n` :)
```suggestion
        if [ -n "$BUILD_HOST" ] && [ -n "$BUILD_HOST_REPLACEMENT" ]; then
```

File: helpers/copy_tree.sh [hunk 3]
```diff
@@ -27,6 +29,20 @@ get_closest() {
...
+            sed -i "s/$BUILD_HOST/$BUILD_HOST_REPLACEMENT/g" "$1"
+        fi
+        if [ ! -z "$BUILD_USER" ] && [ ! -z "$BUILD_USER_REPLACEMENT" ]; then
```
Comment (sledges) [inline helpers/copy_tree.sh:40]: shellcheck prefers this to be `-n` :)
```suggestion
        if [ -n "$BUILD_USER" ] && [ -n "$BUILD_USER_REPLACEMENT" ]; then
```

**Review by sledges — CHANGES_REQUESTED** [review]

File: helpers/copy_tree.sh (hunk 1 above)
Comment (krnlyng) [reply helpers/copy_tree.sh:67]: Is it worth handling possible final slashes (e.g. ~/system/) in this?

File: helpers/copy_tree.sh [hunk 4]
```diff
@@ -46,6 +62,16 @@ sed -i "/$token/,\$d" "$modify_spec"
 echo "$token" >> "$modify_spec"
 echo "%post" >> "$modify_spec"
 
+BUILD_HOST_REPLACEMENT=$(grep "%define build_host_replacement" "$modify_spec" | cut -d ' ' -f3)
+BUILD_USER_REPLACEMENT=$(grep "%define build_user_replacement" "$modify_spec" | cut -d ' ' -f3)
+
+if [ -n "$BUILD_HOST_REPLACEMENT" ] && [[ ${trees[0]} != *system ]]; then
```
Comment (sledges) [inline helpers/copy_tree.sh:68]:
```suggestion
if [ -n "$BUILD_HOST_REPLACEMENT" ] && [[ ${trees[0]//\//} != *system ]]; then
```
sure

Comment (krnlyng) [reply helpers/copy_tree.sh]: yep seems to work

**Review by sledges — APPROVED** [review]

**Pull request merged**